benchmarks/address_manager_store.py

```diff
@@ -0,0 +1,133 @@
+
+        # Construct TimestampedPeerInfo
+        tpi = TimestampedPeerInfo(host=host, port=uint16(port), timestamp=uint64(timestamp))
+
+        # Create the ExtendedPeerInfo
+        epi = ExtendedPeerInfo(
+            addr=tpi,
+            src_peer=None,  # will default to itself inside constructor
+        )
+
+        am.tried_count += 1  # why do we even have `assert tried_ids != tried_count`?
+        node_id = am.id_count
+        am.id_count += 1
+        epi.random_pos = len(am.random_pos)
+        am.map_info[node_id] = epi
+        am.map_addr[epi.peer_info.host] = node_id
+        am.random_pos.append(node_id)
+
+        if i >= num_new:
+            # make a tried_table entry
+            epi.is_tried = True
+            epi.last_success = timestamp
+            epi.last_try = timestamp - rand.randint(0, 1000)
+            bucket = epi.get_tried_bucket(am.key)
+            pos = epi.get_bucket_position(am.key, False, bucket)
+            if am.tried_matrix[bucket][pos] == -1:
+                am.tried_matrix[bucket][pos] = node_id
+                am.tried_count += 1
+        else:
+            # make a new_table entry
+            ref_count = rand.randint(1, NEW_BUCKETS_PER_ADDRESS)
+            epi.ref_count = ref_count
+            assigned = False
+            for _ in range(ref_count):
+                bucket = epi.get_new_bucket(am.key)
+                pos = epi.get_bucket_position(am.key, True, bucket)
+                if am.new_matrix[bucket][pos] == -1:
+                    am.new_matrix[bucket][pos] = node_id
+                    am.new_count += 1
+                    assigned = True
+                    break
+            if not assigned:
+                # fallback if no bucket available
+                epi.ref_count = 0
+
+    return am
+
+
+async def benchmark_serialize_deserialize(iterations: int = 5) -> None:
+    """
+    Benchmarks the serialization and deserialization of peer data.
+    """
+
+    total_serialize_time = 0.0
+    total_deserialize_time = 0.0
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        peers_file_path = Path(tmpdir) / "peers.dat"
+
+        for i in range(iterations):
+            address_manager: AddressManager = populate_address_manager()
+            print(f"--- Benchmark Run {i + 1} ---")
+
+            # Benchmark serialize
+            start_serialize = time.perf_counter()
+            await AddressManagerStore.serialize_bytes(address_manager, peers_file_path)
+            end_serialize = time.perf_counter()
+            serialize_duration = end_serialize - start_serialize
+            total_serialize_time += serialize_duration
+            print(f"Serialize time: {serialize_duration:.6f} seconds")
+
+            # Benchmark deserialize
+            data: Optional[io.BytesIO] = None
+            async with aiofiles.open(peers_file_path, "rb") as f:
+                data = io.BytesIO(await f.read())
+            assert data is not None
```

Comment:
we may do more adjustments around aiofiles and bytesio, but just in terms of hinting and asserts, i think this should be fine.

```suggestion
            async with aiofiles.open(peers_file_path, "rb") as f:
                data = io.BytesIO(await f.read())
```